federatedscope/core/worker/client.py

```diff
@@ -256,7 +256,7 @@ def callback_funcs_for_evaluate(self, message: Message):
             self.trainer.update(message.content)
         metrics = {}
         for split in self._cfg.eval.split:
-            eval_metrics = self.trainer.evaluate(target_data_split_name=split)
+            eval_metrics = self.trainer.evaluate(mode=split, target_data_split_name=split)
```

Comment:
As we want to evaluate train split, I think the run_routine should not be `train`.